security/secvarctl-tests.py

```diff
@@ -31,35 +31,46 @@ def setUp(self):
         Install the basic packages to support secvarctl
         '''
         # Check for basic utilities
+        self.distro_version = distro.detect()
         smm = SoftwareManager()
-        deps = ['gcc', 'make', 'openssl-devel']
+        deps = ['gcc', 'make', 'cmake']
+        if self.distro_version.name in ['rhel', 'redhat']:
+            deps.extend(['openssl-devel'])
+        if self.distro_version.name in ['SuSE']:
+            deps.extend(['libopenssl-devel'])
         for package in deps:
             if not smm.check_installed(package) and not smm.install(package):
                 self.cancel('%s is needed for the test to be run' % package)
+        self.srcdir = ""
         run_type = self.params.get('type', default='upstream')
         if run_type == "upstream":
-            def_url = ("https://github.com/open-power/secvarctl/archive/"
-                       "refs/heads/main.zip")
-            url = self.params.get('url', default=def_url)
-            tarball = self.fetch_asset(url, expire='7d')
-            archive.extract(tarball, self.workdir)
-            self.srcdir = os.path.join(self.workdir, 'secvarctl-main/test')
+            url = "https://github.com/open-power/secvarctl"
+            git.get_repo(url, destination_dir=self.workdir, submodule=True, branch="main")
+            self.srcdir = self.workdir
         elif run_type == "distro":
             self.srcdir = os.path.join(self.workdir, "secvarctl-distro")
             if not os.path.exists(self.srcdir):
                 os.makedirs(self.srcdir)
             self.srcdir = smm.get_source('secvarctl', self.srcdir)
             if not self.srcdir:
                 self.fail("secvarctl source install failed.")
-            self.srcdir = os.path.join(self.srcdir, 'test')
-        os.chdir(self.srcdir)
+        self.build_dir = os.path.join(self.srcdir, "build")
+        os.mkdir(self.build_dir)
+        os.chdir(self.build_dir)
+        rc = process.system("cmake ../")
+        if rc:
+            self.cancel("Can't run 'cmake'")
```

Comment:
Can have some more meaningful message , one suggestion is 
`test need to cancel as running cmake has issue , please check environment `